docker/clang-python3/boost.patch

```diff
@@ -0,0 +1,73 @@
+diff -ru boost/mpi/collectives/all_reduce.hpp boost/mpi/collectives/all_reduce.hpp
+--- boost/mpi/collectives/all_reduce.hpp	2017-12-20 14:26:58.105542000 +0100
++++ boost/mpi/collectives/all_reduce.hpp	2017-12-20 14:56:39.862227286 +0100
+@@ -51,7 +51,8 @@
+                   T* out_values, Op op, mpl::false_ /*is_mpi_op*/,
+                   mpl::true_ /*is_mpi_datatype*/)
+   {
+-    user_op<Op, T> mpi_op(op);
```

Comment:
can you comment on these patch changes?

Reply:
this patch was designed for Clang-tidy (#15, #16), but I don't see any difference in the espresso compilation output in containers built with and without this patch (with static analysis enabled)

Reply:
so we don't need it anymore?

Reply:
I just rebuilt the original `clang:6.0` container with and without the patch, and without the patch the output log is flooded with Clang-tidy warnings. I don't understand why this doesn't happen with `clang-python3:6.0`. The python version should be irrelevant when compiling core C++ files, right?

Reply:
When rebuilding `clang:6.0` locally with the boost patch and sphinx either pinned to version 2.1.1 (the version from the currently built gitlab container) or removed (#106), I get back almost all the Clang-tidy errors that should be removed by the boost patch.

Reply:
I'm removing the patch in `clang-python3:6.0`.

Reply:
how does the boost patch interact with sphinx?

Reply:
no idea, I pinned the version with `pip2 install sphinx=2.1.1` to get the same sphinx version as in the working gitlab `clang:6.0`, and still got most of the clang-tidy messages; also the `clang-python3:6.0` only differs in the python version, but the patch has no effect

Reply:
after discussing the matter with @mkuron, it turned out I didn't use the correct CMake version in some of these experiments: we need CMake v3.8.2 or more recent otherwise the Clang-tidy functionality is silently disabled; the situation returned to normal after a441362 and has always been python-independent; the boost patch is only necessary because Ubuntu 16.04 uses an old boost version